# Individual Work by Hanxing Li(hl3282)

In [1]:
from __future__  import print_function, division
import pylab as pl
import pandas as pd
import numpy as np
import os

%pylab inline

if os.getenv ('PUIDATA') is None:
    print ("Must set env variable PUI2018")


Populating the interactive namespace from numpy and matplotlib




# IDEA:
## The new bus route improves commute.

# NULL HYPOTHESIS:
## the commute time is on average the same or longer with the new bus route as it was before
# ALTERNATIVE HYPOTHESIS: 
## The commute time is on average shorter with the new bus route as it was before

# _$H_0$_ : ${T_{\mathrm{new}}} >= {T_{\mathrm{old}}}$

# _$H_1$_ : ${T_{\mathrm{new}}} < {T_{\mathrm{old}}}$
or identically:

# _$H_0$_ : ${T_{\mathrm{new}}} - {T_{\mathrm{old}}} >= 0$
# _$H_1$_ : ${T_{\mathrm{new}}} - {T_{\mathrm{old}}} < 0$
## I will use a significance level  $\alpha=0.05$

#### which means i want the probability of getting a result at least as significant as mine to be less then 5%

# $ Z = \frac{\mu_{\mathrm{pop}}-\mu_{\mathrm{sample}}}{\sigma/\sqrt{N}} $

In [2]:
!curl https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt > bustime.txt
import os

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1179  100  1179    0     0  10882      0 --:--:-- --:--:-- --:--:-- 10916


In [3]:
os.environ["PUIDATA"] = "%s/PUIdata"%os.getenv("HOME")

print("PUIdata", os.getenv("PUIDATA"))

PUIdata /nfshome/hl3282/PUIdata


In [4]:
!mv bustime.txt $PUIDATA

In [5]:
!ls $PUIDATA

201501-citibike-tripdata.csv  bustime.txt
201501-citibike-tripdata.zip  nyc_water_consumption.csv


In [6]:
import pandas as pd
bustime = pd.read_table("%s/bustime.txt"%os.getenv("PUIDATA"),header = None)
bustime.rename(columns={bustime.columns[0]: "bustime"}, inplace=True)

In [7]:
bustime.head(8)

bustime
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550
5  34.822079
6  39.871884
7  39.579946

In [8]:
bustime.tail(3)

bustime
97  33.731470
98  37.238591
99  36.339841

In [9]:
size = bustime.size

In [10]:
u= bustime['bustime'].mean()
print(u)

34.4661616883


In [11]:
Upop = 36
sigma = 6
Z = (Upop - u )/(sigma/(size)**0.5)
print(Z)

2.55639718617


## Z score = 2.56 > 2, so we are farther than 2 sigmas away from the old mean. Thus, we should reject NULL hypothesis at a significance level of 0.05(the commute time is on average the same or longer with the new bus route as it was before), which confirms my idea: The new bus route improves commute.